icu4c/source/test/intltest/loctest.cpp

```diff
@@ -2748,4 +2751,91 @@ void LocaleTest::TestBug13554() {
     }
 }
 
+void LocaleTest::TestForLanguageTag() {
+    IcuTestErrorCode status(*this, "TestForLanguageTag()");
+
+    static const char tag_en[] = "en-US";
+    static const char tag_oed[] = "en-GB-oed";
+    static const char tag_af[] = "af-t-ar-i0-handwrit-u-ca-coptic-x-foo";
+    static const char tag_ill[] = "!";
+    static const char tag_no_nul[] = { 'e', 'n', '-', 'G', 'B' };
+
+    static const Locale loc_en("en_US");
+    static const Locale loc_oed("en_GB@x=oed");
```

Comment:
We don't turn it into en-GB-oxendict? Bug ticket?

https://www.iana.org/assignments/language-subtag-registry/language-subtag-registry
Type: grandfathered
Tag: en-GB-oed
Description: English, Oxford English Dictionary spelling
Added: 2003-07-09
Deprecated: 2015-04-17
Preferred-Value: en-GB-oxendict

Reply:
It's a known bug. See [ICU bug 9562](https://unicode-org.atlassian.net/browse/ICU-9562) and bugs mentioned there .  See also [ICU bug 13650](https://unicode-org.atlassian.net/browse/ICU-13650) .  @yumaoka 

Chromium has a local patch to fix that and a lot of other discrepancies between the IANA language tag registry and ICU.   I want to upstream them as a PR for [ICU bug 9562]( https://unicode-org.atlassian.net/browse/ICU-9562) .